### PR TITLE
Provider: add deepseek support

### PR DESCRIPTION
**Pull request by nsdou**
**Title:** 
- Support [deepseek](https://www.deepseek.com/) provider

### CODE REVIEW

**Comment by vrushankportkey** [comment]
Thank you for the integration @nsdou! This is terrific

File: src/providers/deepseek/api.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import { ProviderAPIConfig } from '../types';
+
+const DeepSeekAPIConfig: ProviderAPIConfig = {
+  getBaseURL: () => 'https://api.deepseek.com',
+  headers: ({ providerOptions }) => {
+    return { Authorization: `Bearer ${providerOptions.apiKey}` }; // https://platform.deepseek.com/api_keys
```
Comment (narengogi) [inline src/providers/deepseek/api.ts:6]: comment can be removed

**Review by narengogi — APPROVED** [review]
LGTM, verified by making calls to the provider through the gateway
@VisargD

**Comment by VisargD** [comment]
Hey @nsdou - Sorry for the delay on this. Can you please resolve merge conflicts. We will merge the PR post that.

**Pull request merged**